sopel/cli/config.py

```diff
@@ -0,0 +1,161 @@
+
+import argparse
+import os
+
+from sopel import tools, config
+from . import utils
+
+
+def build_parser():
+    """Configure an argument parser for ``sopel-config``"""
+    parser = argparse.ArgumentParser(
+        description='Sopel configuration tool')
+
+    # Subparser: sopel-config <sub-parser> <sub-options>
+    subparsers = parser.add_subparsers(
+        help='Actions to perform',
+        dest='action')
+
+    # sopel-config list
+    list_parser = subparsers.add_parser(
+        'list',
+        help='List available configurations from Sopel\'s homedir',
+        description="""
+        List available configurations from Sopel's homedir ({homedir}) with the
+        extension "{ext}".
+        """.format(homedir=config.DEFAULT_HOMEDIR, ext='.cfg'))
+    list_parser.add_argument(
+        '-e', '--ext', '--extension',
+        dest='extension',
+        default='.cfg',
+        help='Filter by extension (default to "%(default)s)"')
+    list_parser.add_argument(
+        '-p', '--path',
+        action='store_true',
+        dest='display_path',
+        default=False,
+        help='Display a list of absolute filenames instead of their names')
+
+    # sopel-config init
+    init_parser = subparsers.add_parser(
+        'init',
+        help='Initialize sopel configuration file',
+        description='Initialize sopel configuration file')
+    utils.add_common_arguments(init_parser)
+
+    # sopel-config get <section> <key>
+    get_parser = subparsers.add_parser(
+        'get',
+        help='Get a configuration option\'s value',
+        description='Get a configuration option\'s value',
+    )
+    get_parser.add_argument('section')
+    get_parser.add_argument('option')
+    utils.add_common_arguments(get_parser)
+
+    return parser
+
+
+def handle_list(options):
+    """Display a list of configuration available from Sopel's homedir
+
+    :param options: argument parser's parsed option
+
+    This command display an unordered list of config's name from the default
+    Sopel's homedir, without their extension::
+
+        $ sopel-config list
+        default
+        custom
+
+    It is possible to filter by extension using the ``-e/--ext/--extension``
+    option, default to ``.cfg`` (the ``.`` prefix is not required).
+    """
+    display_path = getattr(options, 'display_path', False)
+    extension = getattr(options, 'extension', '.cfg')
+    if not extension.startswith('.'):
+        extension = '.' + extension
+    configs = utils.enumerate_configs(config.DEFAULT_HOMEDIR, extension)
+
+    for config_filename in configs:
+        if display_path:
+            print(os.path.join(config.DEFAULT_HOMEDIR, config_filename))
+        else:
+            name, _ = os.path.splitext(config_filename)
+            print(name)
+
+
+def handle_init(options):
+    """Use config's wizard to initialize a new configuration file for the bot
+
+    :param options: argument parser's parsed options
```

Comment:
If you decide to change the similar param docstring I commented on above, also change this to match. :)